var/spack/repos/builtin/packages/cfdem/package.py

```diff
@@ -0,0 +1,137 @@
+        env.set('CFDEM_TUT_DIR','{}/tutorials'.format(stage_path))
+        env.set('CFDEM_LIGGGHTS_MAKEFILE_POSTIFX','')
+        env.set('CFDEM_VERBOSE','false' )
+        
+
+
+        # Some commands or variables such as wmakeLnInclude  are used when cfdem is compiled, 
+        # and they are defined in etc/bashrc of other package openfoam-org.
+        openfoam_bashrc = join_path(self.spec['openfoam-org'].prefix, 'etc/bashrc')
+        try:
+            env.extend(EnvironmentModifications.from_sourcing_file(
+                openfoam_bashrc, clean=True
+            ))
+        except Exception as e:
+            msg = 'unexpected error when sourcing openfoam-org bashrc [{0}]'
+            tty.warn(msg.format(e))
+
+        # After importing openfoam-org's bashrc, the variable WM_PROJECT_VERSION will be set to 5.0,
+        # but the format required by cfdem is 5.x
+        of_version = '5.x'
+        if self.spec['openfoam-org'].version == Version("5.0"):
+            of_version = '5.x'
+        elif self.spec['openfoam-org'].version == Version("3.0.1"):
+            of_version = '3.0.x'
+
+        env.set('WM_PROJECT_VERSION', of_version)
+
+        # liggghts header files and libraries
+        liggghts_pre = self.spec['liggghts'].prefix
+        env.set('CFDEM_LIGGGHTS_SRC_DIR',  join_path(liggghts_pre, 'src'))
+        env.set('CFDEM_LIGGGHTS_LIB_PATH',  join_path(liggghts_pre, 'src'))
+
+
+    def setup_run_environment(self, env):
+
+        env.set('CFDEM_SRC_DIR','{}/src'.format(self.prefix))
+        
+
+        with working_dir('./src/lagrangian/cfdemParticle/etc/'):
+            with open('exec_all.sh', 'w') as f:
+                f.write('#!/bin/bash\n')
+                f.write('\n')
+                f.write('echo y >/tmp/autoinput\n')
+                f.write('source $CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/bashrc < /tmp/autoinput\n')
+                f.write('\n')
+                f.write('. $CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/compileCFDEMcoupling_src.sh\n')
+                f.write('echo "#################### finished compileCFDEMcoupling_src ####################"\n')
+                f.write('\n')
+                f.write('. $CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/compileCFDEMcoupling_sol.sh\n')
+                f.write('echo "#################### finished compileCFDEMcoupling_sol ####################"\n')
+                f.write('\n')
+                f.write('. $CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/compileCFDEMcoupling_uti.sh\n')
+                f.write('echo "#################### finished compileCFDEMcoupling_uti ####################"\n')
+                f.write('\n')
+
+    def build(self, spec, prefix):
+
+        # run shell script exec_all.sh to compile
+        shscript = join_path(os.getcwd(), 'src/lagrangian/cfdemParticle/etc/exec_all.sh')
+        set_executable(shscript)
+        script = Executable(shscript)
+        script()
```

Comment:
Where is the script defined at build time? Is it different than the one that is written at run time?